packages/dds/tree/src/feature-libraries/chunked-forest/chunkTree.ts

```diff
@@ -226,75 +231,149 @@ export function makePolicy(policy?: Partial<ChunkPolicy>): ChunkPolicy {
 	return withDefaults;
 }
 
-export function shapesFromSchema(
-	schema: StoredSchemaCollection,
-	policy: FullSchemaPolicy,
-): Map<TreeNodeSchemaIdentifier, ShapeInfo> {
-	const shapes: Map<TreeNodeSchemaIdentifier, ShapeInfo> = new Map();
-	for (const identifier of schema.nodeSchema.keys()) {
-		tryShapeFromSchema(schema, policy, identifier, shapes);
-	}
-	return shapes;
+export interface ShapeFromSchemaParameters {
+	/**
+	 * The collection of stored schemas containing node and field definitions.
+	 */
+	readonly schema: StoredSchemaCollection;
+	/**
+	 * Policy from the app for interpreting the stored schema.
+	 */
+	readonly policy: FullSchemaPolicy;
```

Comment:
We already have SchemaAndPolicy type which you can extend.

```suggestion
export interface ShapeFromSchemaParameters extends SchemaAndPolicy {
```

That stores a SchemaPolicy instead of a FullSchemaPolicy, but I think that should be fine for this usecase.